moveit_core/python/tools/CMakeLists.txt

```diff
@@ -1,13 +1,16 @@
 set(MOVEIT_LIB_NAME moveit_python_tools)
 
-find_package(Python COMPONENTS Interpreter Development QUIET)
-
 add_library(${MOVEIT_LIB_NAME}
   src/pybind_rosmsg_typecasters.cpp
 )
 add_dependencies(${MOVEIT_LIB_NAME} ${catkin_EXPORTED_TARGETS})
+target_include_directories(${MOVEIT_LIB_NAME} SYSTEM PRIVATE ${PYTHON_INCLUDE_DIRS} ${pybind11_INCLUDE_DIRS})
 target_link_libraries(${MOVEIT_LIB_NAME} ${catkin_LIBRARIES} ${PYTHON_LIBRARIES})
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+  # avoid "no matching function for call to 'operator delete'" - https://github.com/pybind/pybind11/issues/1604
+  target_compile_options(${MOVEIT_LIB_NAME} PUBLIC -fsized-deallocation)
+endif()
```

Comment:
This issue was fixed in https://github.com/pybind/pybind11/commit/759221f5c56939f59d8f342a41f8e2d2cacbc8cf, i.e. version 2.5.0.

Reply:
I found it necessary during development of the patch. I agree though that this shouldn't be necessary. I reverted the patch now that everything passes. Let's see.

Reply:
Please see CI for the latest commit, the error persists without the clang override.

Reply:
Hence, I was asking to keep `pybind11_catkin`, which provides version 2.5.0 for Noetic right now.